stable-profiles/0.9.52/firejailed-tor-browser.profile

```diff
@@ -0,0 +1,94 @@
+# Firejail profile for firejailed-tor-browser
+# This file is overwritten after every install/update
+# Persistent local customizations
+include ${HOME}/.config/firejail/firejailed-tor-browser.local
+# Persistent global definitions
+include /etc/firejail/globals.local
+
+# Note: PluggableTransports didn't work with this profile
+
+noblacklist ${HOME}/.firejailed-tor-browser
+
+blacklist /opt
+blacklist /run/dbus/system_bus_socket
+blacklist /srv
+blacklist /usr/games
+blacklist /usr/local
+blacklist /usr/src
+
+include /etc/firejail/disable-common.inc
+include /etc/firejail/disable-devel.inc
+include /etc/firejail/disable-passwdmgr.inc
+include /etc/firejail/disable-programs.inc
+
+whitelist ${HOME}/.firejailed-tor-browser
+whitelist /usr/share/alsa
+whitelist /usr/share/applications
+whitelist /usr/share/ca-certificates
+whitelist /usr/share/ca-certificates-java
+whitelist /usr/share/dconf
+whitelist /usr/share/distro-info
+whitelist /usr/share/drirc.d
+whitelist /usr/share/enchant
+whitelist /usr/share/fontconfig
+whitelist /usr/share/fonts
+whitelist /usr/share/glib-2.0
+whitelist /usr/share/glvnd
+whitelist /usr/share/gtksourceview-3.0
+whitelist /usr/share/hunspell
+whitelist /usr/share/icons
+whitelist /usr/share/knotifications5
+whitelist /usr/share/kservices5
+whitelist /usr/share/kxmlgui5
+whitelist /usr/share/libdrm
+whitelist /usr/share/libthai
+whitelist /usr/share/locale
+whitelist /usr/share/mime
+whitelist /usr/share/misc
+whitelist /usr/share/p11-kit
+whitelist /usr/share/pixmaps
+whitelist /usr/share/qt5
+whitelist /usr/share/sounds
+whitelist /usr/share/tcltk/
+whitelist /usr/share/terminfo
+whitelist /usr/share/themes
+whitelist /usr/share/X11
+whitelist /usr/share/xml
+whitelist /usr/share/zoneinfo
```

Comment:
Removing all is more permissive. If `blacklist /usr/share` works add it if not add something like
```
whitelsit /usr/share/fonts
whitelist /usr/share/X11
```

When this is done I merge.

Reply:
If by "permissive" you mean "opposite of being strict" then, that is not what I prefer. Thanks for mentioning this very important point and for the suggestions.

When I added `blacklist /usr/share` I noticed the following warning but the browser did start up and I was able to navigate to a webpage.  However, when I tried to open a file I received lots of Gtk errors, the browser aborts, and terminates with a Segmentation fault.:
```
Launching './Browser/start-tor-browser --detach --debug'...
/etc/magic, 0: Warning: using regular magic file `/usr/share/misc/magic'
file: cannot read magic file `/usr/share/misc/magic' (Permission denied)
```
```
(firefox:82): Gtk-WARNING **: 05:46:14.069: Could not find the icon 'user-home-symbolic-ltr'. The 'hicolor' theme
was not found either, perhaps you need to install it.
You can get a copy from:
	http://icon-theme.freedesktop.org/releases

(firefox:82): Gtk-WARNING **: 05:46:14.069: Could not load a pixbuf from /org/gtk/libgtk/icons/16x16/status/image-missing.png.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /org/gtk/libgtk/icons/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Redirecting call to abort() to mozalloc_abort

Exiting due to channel error.
Exiting due to channel error.
Oct 09 05:46:14.000 [notice] Owning controller connection has closed -- exiting now.
Oct 09 05:46:14.000 [notice] Catching signal TERM, exiting cleanly.
Exiting due to channel error.
./Browser/start-tor-browser: line 371:    82 Segmentation fault      (core dumped) TOR_CONTROL_PASSWD=${TOR_CONTROL_PASSWD} ./firefox --class "Tor Browser" -profile TorBrowser/Data/Browser/profile.default "${@}" < /dev/null

Parent is shutting down, bye...
```
So, the browser appears to be working with this blacklist statement but I do not like that warning.

Next, I removed `blacklist /usr/share` and tried:
```
whitelist /usr/share/fonts
whitelist /usr/share/X11
```
And I noticed a similar warning but the browser did start up and I was able to navigate to a webpage:
```
Launching './Browser/start-tor-browser --detach --debug'...
/etc/magic, 0: Warning: using regular magic file `/usr/share/misc/magic'
file: could not find any valid magic files!
```
So then I tried:
```
whitelist /usr/share/fonts
whitelist /usr/share/misc
whitelist /usr/share/X11
```
This time there was no more warning about `/usr/share/misc/magic`. However, when I tried to open a file I received lots of Gtk errors, the browser aborts, and terminates with a Segmentation fault as previously shown above.

In conclusion, I'm adding all those whitelist statements I previously removed. Your point is clear to me now, those statements are necessary and are indeed less permissive than without them. Thank you.

Reply:
> If by "permissive" you mean "opposite of being strict" then, that is not what I prefer.

If you have no `whitelist /usr/share/foobar`, all files in /usr/share are inside the sandbox. If you have at least one `whitelist /usr/share/foobar`, only files/dirs with a whitelist command are inside the sandbox.

Reply:
Oh ok, I get it now. That makes allot more sense to me and it certainly explains my results. Thank you for the clarification.